docs/STYLE_GUIDE.md

```diff
@@ -0,0 +1,123 @@
+  `Vec<T>` must have the same length), then add a paragraph explaining this.
+- Begin every docstring with a meaningful one sentence description of the
+  function in third person.
+- Avoid WET documentation such as this:
+
+  ```rust
+  /// Rejects a market.
+  ///
+  /// # Arguments:
+  ///
+  /// - `origin`: The origin calling the dispatchable
+  /// - `market_id`: The market id of the market to reject
+  pub fn reject_market(
+      origin: OriginFor<T>,
+      #[pallet::compact] market_id: MarketIdOf<T>,
+  ) -> DispatchResultWithPostInfo {
+      // --- snip! ---
+  }
+  ```
+
+  Self-explanatory function arguments need not be documented if they are
+  explained in full text. Instead, focus on describing the effects of the
+  function. Someone who doesn't know how to read the code must be able to
+  understand what the function does.
+
+- Docstrings for dispatchable **should** contain a section on runtime complexity
+  (to make understanding the benchmarks easier).
+- Docstrings for dispatchables need not document the `origin` parameter, but
+  should specify what origins the dispatchable may be called by.
+- Docstrings for dispatchables **must** include the events that the dispatchable
+  emits.
+- Use `#![doc = include_str!("../README.md")]`.
+- Detail non-trivial algorithms in comments inside the function.
+
+An example of a good docstring would be this:
+
+```rust
+/// Rejects a market, destroying the market and unreserving and/or slashing the creator's advisory
+/// bond.
+///
+/// May only be (successfully) called by `RejectOrigin`. The fraction of the advisory bond that is
+/// slashed is determined by `AdvisoryBondSlashPercentage`.
+pub fn reject_market(
+    origin: OriginFor<T>,
+    #[pallet::compact] market_id: MarketIdOf<T>,
+) -> DispatchResultWithPostInfo {
+    // --- snip! ---
+}
+```
+
+Info like "The fraction of the advisory bond that is slashed is determined by
+`AdvisoryBondSlashPercentage`" **may** be stored in `README.md` to avoid
+duplicating documentation.
+
+## Formatting
+
+- rustfmt and clippy have the last say in formatting.
+- Please format code contained in macro invocations (`impl_benchmarks!`,
+  `decl_runtime_apis!`, homebrew macros in `runtime/`, etc.) and attributes
+  (`#[pallet::weight(...)`, etc.) manually.
+- Add trailing commas in macro invocations manually, as rustfmt won't add them
+  automatically.
+
+  ```rust
+  ensure!(
+      a_very_very_very_very_very_very_very_long_variable,
+      b_very_very_very_very_very_very_very_long_variable, // This comma is not ensured by rustfmt.
+  )
+  ```
+
+## Code Style
+
+- Never use panickers.
+- Prefer double turbofish `Vec::<T>::new()` over single turbofish
+  `<Vec<T>>::new()`.
+- All branches of match expressions must be explicit. Never use the catch-all
+  `_ =>`. For example:
+
+  ```rust
+  match market_status {
+      MarketStatus::Active => {
+          // --- snip ---
+      }
+      _ => {
+          // --- snip ---
+      } // not allowed
```

Comment:
I personally wouldn't be too strict here. There is a reason why this catch all pattern exists. Current example: I don't want to write all enum variants to [this match](https://github.com/zeitgeistpm/zeitgeist/blob/b6400f6016854e6b1f1d59c1473e1a352db5c200/runtime/common/src/lib.rs#L1026).